test/unit/ssz_ex_test.exs

```diff
@@ -22,4 +22,27 @@ defmodule Unit.SSZExTest do
     assert_roundtrip(<<1>>, true, :bool)
     assert_roundtrip(<<0>>, false, :bool)
   end
+
+  test "serialize and deserialize list" do
+    assert_roundtrip(<<1, 1, 0>>, [true, true, false], {:list, :bool, 3})
+
+    assert_roundtrip(
+      <<230, 0, 0, 0, 124, 1, 0, 0, 11, 2, 0, 0>>,
+      [230, 380, 523],
+      {:list, {:int, 32}, 3}
+    )
+
+    variable_list = [[5, 6], [7, 8], [9, 10]]
+    encoded_variable_list = <<12, 0, 0, 0, 14, 0, 0, 0, 16, 0, 0, 0, 5, 6, 7, 8, 9, 10>>
+    assert_roundtrip(encoded_variable_list, variable_list, {:list, {:list, {:int, 8}, 2}, 3})
```

Comment:
We should add some tests with a list length that differs from the schema's max length.